File: articles/psychological-types-introversion-extraversion-and-the-four-functions/go/type_dynamics.go
package main

import "fmt"

func compensationStrain(
	dominantStrength,
	inferiorStrength,
	functionVariance,
	unconsciousPressure,
	reflectiveCapacity float64,
) float64 {
	gap := dominantStrength - inferiorStrength
	return 0.70*gap +
		0.55*functionVariance +
		0.45*unconsciousPressure -
		0.28*reflectiveCapacity
}

func developmentalIntegration(
	inferiorStrength,
	symbolicRelation,
	reflectiveCapacity,
	functionVariance,
	strain float64,
) float64 {
	return 0.42*inferiorStrength +
		0.38*symbolicRelation +
		0.36*reflectiveCapacity -
		0.32*functionVariance -
		0.26*strain
}

func main() {
	strain := compensationStrain(0.92, 0.78, 0.004, 0.20, 0.86)
	integration := developmentalIntegration(0.78, 0.84, 0.86, 0.004, strain)

	fmt.Printf("Synthetic compensation strain: %.3f\n", strain)
	fmt.Printf("Synthetic developmental integration score: %.3f\n", integration)
}

File: articles/psychological-types-introversion-extraversion-and-the-four-functions/rust/type_dynamics.rs
fn compensation_strain(
    dominant_strength: f64,
    inferior_strength: f64,
    function_variance: f64,
    unconscious_pressure: f64,
    reflective_capacity: f64,
) -> f64 {
    let gap = dominant_strength - inferior_strength;
    0.70 * gap
        + 0.55 * function_variance
        + 0.45 * unconscious_pressure
        - 0.28 * reflective_capacity
}

fn developmental_integration(
    inferior_strength: f64,
    symbolic_relation: f64,
    reflective_capacity: f64,
    function_variance: f64,
    strain: f64,
) -> f64 {
    0.42 * inferior_strength
        + 0.38 * symbolic_relation
        + 0.36 * reflective_capacity
        - 0.32 * function_variance
        - 0.26 * strain
}

fn main() {
    let strain = compensation_strain(0.92, 0.78, 0.004, 0.20, 0.86);
    let integration = developmental_integration(0.78, 0.84, 0.86, 0.004, strain);

    println!("Synthetic compensation strain: {:.3}", strain);
    println!("Synthetic developmental integration score: {:.3}", integration);
}
